src/Components/Components/src/Rendering/Renderer.cs

```diff
@@ -452,8 +452,29 @@ private void ProcessRenderQueue()
             }
         }
 
-        private Task InvokeRenderCompletedCalls(ArrayRange<RenderTreeDiff> updatedComponents)
+        private Task InvokeRenderCompletedCalls(ArrayRange<RenderTreeDiff> updatedComponents, Task updateDisplayTask)
         {
+            if (updateDisplayTask.IsCanceled)
+            {
+                // The display update was cancelled (maybe due to a timeout on the components server-side case or due
+                // to the renderer being disposed)
+                return Task.CompletedTask;
+            }
+            if (updateDisplayTask.IsFaulted)
+            {
+                // The display update failed so we don't care any more about running on render completed
+                // fallbacks as the entire rendering process is going to be torn down.
+                HandleException(updateDisplayTask.Exception);
+                return Task.CompletedTask;
+            }
+
+            if (!updateDisplayTask.IsCompleted)
+            {
+                var copy = new RenderTreeDiff[updatedComponents.Count];
+                Array.Copy(updatedComponents.Array, 0, copy, 0, updatedComponents.Count);
+                return updateDisplayTask.ContinueWith(t => InvokeRenderCompletedCalls(new ArrayRange<RenderTreeDiff>(copy, copy.Length), t));
```

Comment:
Shouldn't this be kept on the renderer's sync context? `ContinueWith` will not retain the sync context.

Reply:
You're right

Reply:
Done